.github/workflows/release.yaml

```diff
@@ -17,4 +17,4 @@ jobs:
           gh auth login --with-token <<<'${{ secrets.GITHUB_TOKEN }}'
           docker login docker.io --username ${{ secrets.DOCKERHUB_USER }} --password-stdin <<<'${{ secrets.DOCKERHUB_TOKEN }}'
       - name: release
-        run: make release DOCKER=1 RELEASE_TAG=${{ github.event.inputs.tag }}
+        run: make -f builder/Makefile.release release RELEASE_TAG=${{ github.event.inputs.tag }} PUBLISH=y
```

Comment:
I think that fundamentally the release script or release Make target **should not** create nor push Git tags to the upstream or your forks.

Creating and pushing an annotated Git tag to the upstream should trigger GitHub release workflow that eventually creates GitHub release and publish container images to DockerHub, ECR Gallery, etc.

```
git tag -a v0.7.0 -m 'Release v0.7.0' && git push upstream v0.7.0
````

What's more, we must have a way to test release artifacts that we're about to publish.

To do that release scripts may support two modes run in two different scenarios:
1. [PR Validation Workflow] Release snapshot to make sure that we're ready to release anytime from the main branch and use snapshot build artifacts (i.e. executable binaries, signatures, container images) for integration tests:
   ```
   make -f builder/Makefile.release SNAPSHOT=1
   ```
   The SNAPSHOT=1 flag is here to say that we build snapshot release artifacts with the current Git SHA-1 not the latest Git tag
2. [Release Workflow] Do actual release, i.e. publish changes log, artifacts and container images:
    ```
   make -f builder/Makefile.release PUBLISH=1
   ```

Reply:
Why have our release script double as a builder for integration tests instead of just have a separate integration test script? We could run integration tests at the start of the release script, but seems like it'd be better to separate that.

Reply:
Release snapshot won't double as a builder for integration or e2e test. It's actually a prerequisite to run e2e tests. The idea is to build (expensive) container images only once and reuse them for e2e tests. Otherwise our CI workflow won't be very efficient. More importantly, Make targets and other build scripts are error prone and fragile and probably the only opportunity to test them out (except the publish step) is to build such release snapshot in the PR validation workflow.

Since we plan to add basic smoke test to the PR validation workflow we could reuse container images cached in the previous steps and avoid duplicating build logic in the test runner script. I hope it explains better now.